server/src/main/java/org/elasticsearch/action/search/SearchTransportService.java

```diff
@@ -456,12 +399,13 @@ public static void registerRequestHandler(TransportService transportService, Sea
             SearchFreeContextResponse::readFrom
         );
 
-        transportService.registerRequestHandler(
-            FREE_CONTEXT_ACTION_NAME,
-            freeContextExecutor,
-            SearchFreeContextRequest::new,
-            freeContextHandler
-        );
+        // TODO: remove this handler once the lowest compatible version stops using it
```

Comment:
can you make this an assert or at least clarify when it can be removed?

Reply:
Right, will do after backporting!